src/link/api.js

```diff
@@ -0,0 +1,121 @@
+let abortFunctions = [];
+
+export const prefixSearch = ( lang, term, callback ) => {
+	const params = {
+		action: 'query',
+		prop: 'description|pageimages|pageprops',
+		piprop: 'thumbnail',
+		pilimit: 5,
+		ppprop: 'displaytitle',
+		generator: 'prefixsearch',
+		redirects: true,
+		pithumbsize: 64,
+		gpslimit: 5,
+		gpsnamespace: 0,
+		gpssearch: term.replace( /:/g, ' ' ),
+	};
+
+	const url = buildMwApiUrl( lang, params );
+	return request( url, ( data ) => {
+		if ( ! data.query?.pages ) {
+			callback( [] );
+		} else {
+			data.query.pages.sort( ( a, b ) => a.index - b.index );
+			callback(
+				Object.values( data.query.pages ).map( ( page ) => {
+					return {
+						title: page.title,
+						description: page.description,
+						thumbnail: page.thumbnail?.source,
+					};
+				} )
+			);
+		}
+	} );
+};
+
+export const fulltextSearch = ( lang, term, callback ) => {
+	const params = {
+		action: 'query',
+		list: 'search',
+		srprop: 'snippet',
+		srsearch: term,
+		srnamespace: 0,
+		srlimit: 5,
+		srenablerewrites: true,
+		srinfo: 'rewrittenquery',
+		prop: 'pageimages',
+		piprop: 'thumbnail',
+		pithumbsize: 64,
+		pilimit: 5,
+		generator: 'search',
+		gsrsearch: term,
+		gsrnamespace: 0,
+		gsrlimit: 5,
+	};
+
+	const url = buildMwApiUrl( lang, params );
+	return request( url, ( data ) => {
+		if ( ! data.query?.search ) {
+			callback( [] );
+		} else {
+			const { search, pages } = data.query;
+			callback(
+				Object.values( search ).map( ( item ) => {
+					const page =
+						pages &&
+						pages.find( ( { pageid } ) => pageid === item.pageid );
+					return {
+						title: item.title,
+						description: stripHtml( item.snippet ),
+						thumbnail: page.thumbnail?.source,
+					};
+				} )
+			);
+		}
+	} );
+};
+
+const defautParams = {
+	format: 'json',
+	formatversion: 2,
+	origin: '*',
+};
+
+const buildMwApiUrl = ( lang, params ) => {
+	params = Object.assign( {}, defautParams, params );
+	const baseUrl = `https://${ lang }.wikipedia.org/w/api.php`;
+	return (
+		baseUrl +
+		'?' +
+		Object.keys( params )
+			.map( ( p ) => {
+				return `${ p }=${ encodeURIComponent( params[ p ] ) }`;
+			} )
+			.join( '&' )
+	);
+};
+
+const request = ( url, callback ) => {
+	abortFunctions.forEach( ( x ) => x && x.abort() );
+	abortFunctions = [];
+
+	// eslint-disable-next-line no-undef
```

Comment:
I think the proper way to silent this warning is by telling eslint that this code will run in a browser so it understands that `XMLHttpRequest`, `document`, `window` etc are globals. 


In `.eslintrc`:
```
"env": {
        "browser": true
    }
```